### PR TITLE
13021 nist elr java class

### PR DESCRIPTION
**Pull request by jack-h-wang**
This PR adds a new ORU_R01 AbstractMessage implementation for the NIST ELR 2.5.1 spec.

Test Steps:
1. CI should be enough
2. Verify that the classes match what is in the NIST ELR profile https://hl7v2-elr-testing.nist.gov/mu-elr/

## Changes
- Add NIST ELR 2.5.1 ORU_R01 message java class
- Add tests

### Testing
- [ ] Tested locally?
- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [x] Added tests?

## Linked Issues
- #13021

### CODE REVIEW

File: prime-router/src/main/java/fhirengine/translation/hl7/structures/nistelr251/datatype/CE.java [hunk 1]
```diff
@@ -0,0 +1,208 @@
+/*
+ * This class is an auto-generated source file for a HAPI
+ * HL7 v2.x standard structure class.
+ *
+ * For more information, visit: http://hl7api.sourceforge.net/
```
Comment (arnejduranovic) [inline prime-router/src/main/java/fhirengine/translation/hl7/structures/nistelr251/datatype/CE.java:5]: We don't need full docs now, but we should update the PR description to include steps on how you generated these (what inputs and tools did you use, etc) so we can review that process as well.

File: prime-router/src/main/java/fhirengine/translation/hl7/structures/nistelr251/datatype/CE.java [hunk 2]
```diff
@@ -0,0 +1,208 @@
+/*
```
Comment (arnejduranovic) [inline prime-router/src/main/java/fhirengine/translation/hl7/structures/nistelr251/datatype/CE.java:1]: nit: these are in the translation folder but have nothing to do with translation proper?

File: prime-router/src/test/kotlin/fhirengine/translation/NistElrTests.kt [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import ca.uhn.hl7v2.util.Hl7InputStreamMessageIterator
+import ca.uhn.hl7v2.validation.impl.ValidationContextFactory
+import kotlin.test.Test
+import fhirengine.translation.hl7.structures.nistelr251.message.ORU_R01 as NIST_ELR_ORU_R01
+
+class NistElrTests {
+    @Test
+    fun `test nist elr java class`() {
+        val rawMessage = """
+MSH|^~\&|CDC PRIME - Atlanta, Georgia (Dekalb)^2.16.840.1.114222.4.1.237821^ISO|Avante at Ormond Beach^10D0876999^CLIA|PRIME_DOH|Prime ReportStream|20210210170737||ORU^R01^ORU_R01|371784|P|2.5.1|||NE|NE|USA||||PHLabReportNoAck^ELR_Receiver^2.16.840.1.113883.9.11^ISO
```
Comment (arnejduranovic) [inline prime-router/src/test/kotlin/fhirengine/translation/NistElrTests.kt:18]: I'm assuming this message contains all the possible fields the classes support? How was it created?

**Comment by arnejduranovic** [comment]
Also, where is this AC criterion met? 

`A class that extends AbstractMessage exactly matches the NIST ELR 2.5.1 conformance profile`

File: prime-router/src/main/java/fhirengine/translation/hl7/structures/nistelr251/datatype/CE.java (hunk 1 above)
Comment (mkalish) [reply prime-router/src/main/java/fhirengine/translation/hl7/structures/nistelr251/datatype/CE.java:5]: So I'm not sure if I would recommend doing that moving forward?  Maybe we can do a quick 16th on this.

I'm also realizing that we should confirm that these actually differ from the base CE

File: prime-router/src/test/kotlin/fhirengine/translation/NistElrTests.kt (hunk 3 above)
Comment (mkalish) [reply prime-router/src/test/kotlin/fhirengine/translation/NistElrTests.kt:18]: I _think_ Jack grabbed it out of one of the elims messages, but I can take a second look and make sure it's comprehensive

**Comment by mkalish** [comment]
@arnejduranovic That AC is met by the ORU_R01.java file

File: prime-router/src/main/java/fhirengine/translation/hl7/structures/nistelr251/datatype/CE.java (hunk 2 above)
Comment (mkalish) [reply prime-router/src/main/java/fhirengine/translation/hl7/structures/nistelr251/datatype/CE.java:1]: Not sure I follow, this is the same package with the `HL7ToFhirTranslator`?

**Comment by mkalish** [comment]
@arnejduranovic This is ready for another review

**Comment by brick-green** [comment]
The only missing datatype I see is `CWE_ELR_OBX-5`. But the only difference with `CWE_ELR` are in the predicate column in the nist profile which don't impact the java class. Is that why you didn't include it?

**Review by arnejduranovic — APPROVED** [review]

**Comment by mkalish** [comment]
@brick-green I think it didn't get picked up by the code generator and then when I went through the segment's it didn't appear (because OBX-5 varies).  Looking at it now, I think we can safely not create a separate file since the shape is the same?

File: .github/actions/sonarcloud/action.yml [hunk 4]
```diff
@@ -21,7 +20,7 @@ runs:
         SONAR_TOKEN: ${{ inputs.sonar-token }}
       with:
         args: >
-          -Dsonar.coverage.exclusions=prime-router/src/test/**,prime-router/src/testIntegration/**,prime-router/src/main/kotlin/cli/tests/**,frontend-react/**/__mocks__/**,frontend-react/**/mocks/**,frontend-react/**/*.test.*
+          -Dsonar.coverage.exclusions=prime-router/src/test/**,prime-router/src/testIntegration/**,prime-router/src/main/kotlin/cli/tests/**,frontend-react/**/__mocks__/**,frontend-react/**/mocks/**,frontend-react/**/*.test.*,prime-router/src/main/java/**
```
Comment (snesm) [inline .github/actions/sonarcloud/action.yml:23]: Why exclude the 51 java files in `prime-router/src/main/java/**` ?

Comment (mkalish) [reply .github/actions/sonarcloud/action.yml:23]: These are pretty just going to be POJO files/configuration and there is limited value in them getting scanned.  I don't feel super strongly about it though, so I'm happy to add a follow up ticket that will set up sonar to scan java files (it wasn't working off the bat which is why I excluded them)

**Comment by brick-green** [comment]
> @brick-green I think it didn't get picked up by the code generator and then when I went through the segment's it didn't appear (because OBX-5 varies). Looking at it now, I think we can safely not create a separate file since the shape is the same?

Yeah I agree

File: .github/actions/sonarcloud/action.yml (hunk 4 above)
Comment (snesm) [reply .github/actions/sonarcloud/action.yml:23]: This only excludes them from coverage statistics.  Maybe the coverage supporting files are not generated for this new folder.  Please ticket for investigation.

**Review by snesm — APPROVED** [review]

**Pull request merged**